unstructured/partition/doc.py

```diff
@@ -63,52 +63,74 @@ def partition_doc(
         This information will be reflected in elements' metadata and can be be especially
         useful when partitioning a document that is part of a larger document.
     """
-    # Verify that only one of the arguments was provided
-    if filename is None:
-        filename = ""
     exactly_one(filename=filename, file=file)
 
-    if len(filename) > 0:
-        _, filename_no_path = os.path.split(os.path.abspath(filename))
-        base_filename, _ = os.path.splitext(filename_no_path)
```

Comment:
These two lines are part of computing the target "converted" file-path so that path can be sent to `partition_docx()`. They appear both here in the "filename" branch and below in the "file" branch. Move these to just above where that value is used and the source filename is resolved.